app/controllers/two_factor_authentication/personal_key_verification_controller.rb

```diff
@@ -21,16 +25,24 @@ def create
 
     def handle_result(result)
       if result.success?
-        re_encrypt_profile_recovery_pii if password_reset_profile.present?
+        generate_new_personal_key
         handle_valid_otp
       else
         handle_invalid_otp(type: 'personal_key')
       end
     end
 
+    def generate_new_personal_key
+      if password_reset_profile.present?
+        re_encrypt_profile_recovery_pii
+      else
+        user_session[:personal_key] = PersonalKeyGenerator.new(current_user).create
```

Comment:
I still disagree with writing the personal key to the session. We intentionally do not save the personal key anywhere since it can be used to bypass 2FA on sign in and decrypt the user's PII.

While the personal key would be encrypted, it would be encrypted with a shared server secret meaning anybody with access to redis and that secret could read the personal keys of users when it is in the session.

Reply:
Though now I see your comment on the other issue where we do in fact already write it to the session. With that in mind this may be okay with the understanding that we should come back and revisit it here as well as in those places.

Reply:
Yeah, I agree. I will modify this to store a flag in the session instead. I commented in the GitHub issue.

Reply:
Given that our current design requires us to store the key in the session to prevent CSRF attacks, is this PR good to go?

Reply:
Yeah, let me take a last look over the PR